tests/test_linear_ancilla_composite.py

```diff
@@ -0,0 +1,119 @@
+#  Copyright 2024 D-Wave Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+import collections
+import unittest
+
+import dimod
+import networkx as nx
+import numpy as np
+
+from dwave.system.testing import MockDWaveSampler
```

Comment:
The MockSampler can be amended so as to emulate flux_biases as Ising model linear biases:
https://github.com/AndyZzzZzzZzz/shimming-tutorial/blob/andy/tutorial_code/helpers/sampler_wrapper.py

I recommend you use a wrapper like this one and then test optimization using MockSampler().sample

Reply:
The tests in `test_linear_ancilla_composite` do not check for sample correctness. Just for the implementation of ancillas and flux biases in the call. So the MockSampler works for it.

The error that I posted yesterday was related to circle CI testing the example in a [docstring](https://github.com/dwavesystems/dwave-system/pull/530/files#diff-558dca34f40df626e648cc5dc9cb3f09dcafe585a4e5860a540228d64b33a847R44). In that case I am not sure it's possible to wrap with this fancier mock sampler. If it's not trivial to do so I would skip